index.js

```diff
@@ -344,6 +345,45 @@ app.post('/api/doorTest', async(req, res) => {
     await handleSensorRequest(locationid)
 });
 
+app.post('/api/door', async(request, response) => {
+    const coreId = request.body.coreid
+    const locationid = db.getLocationFromCoreId(coreId)
+  
+    const message = JSON.parse(request.body.data)
+    const signal = message.data
+    var doorSignal
+  
+    if (signal==IM21_DOOR_STATUS.OPEN){
+        doorSignal = "open"
+    }
+    else if (signal==IM21_DOOR_STATUS.CLOSED){
+        doorSignal = "closed"
+    }
+    else if (signal==IM21_DOOR_STATUS.LOW_BATT){
+        doorSignal = "LowBatt"
+    }
+    else if (signal==IM21_DOOR_STATUS.HEARTBEAT_OPEN || signal==IM21_DOOR_STATUS.HEARTBEAT_CLOSED){
+        doorSignal = "HeartBeat"
+    }
+  
+    redis.addIM21DoorSensorData(locationid, doorSignal);
+    handleSensorRequest(locationid)
+    response.status(200).json("OK")
+});
+
+// Handler for device vitals such as wifi strength
+app.post('api/devicevitals', async(request, response) => {
```

Comment:
The other endpoint definitions all start with a forward slash:
`app.post('/api/devicevitals', ...` 
would be more consistent.